prow/config/cache.go

```diff
@@ -20,14 +20,76 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"strings"
 	"time"
 
+	"github.com/hashicorp/golang-lru/simplelru"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/sirupsen/logrus"
 
 	"k8s.io/test-infra/prow/cache"
 	"k8s.io/test-infra/prow/git/v2"
 )
 
+// MkEvictCallback constructs a simplelru.EvictCallback function to be used with
+// an InRepoConfigCache. It requires the name of the component and the evictions
+// metric to be used for the callback.
+func MkEvictCallback(component string) simplelru.EvictCallback {
+	evictionsMetric := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: fmt.Sprintf("%s_inRepoConfigCache_evictions", component),
+		Help: "Count of cache evictions by org and repo.",
+	}, []string{
+		"component",
+		"org",
+		"repo",
+	})
+
+	prometheus.MustRegister(evictionsMetric)
+
+	// Every time we evict a key, record it as a Prometheus metric. This way, we
+	// can monitor how frequently evictions are happening (if it's happening too
+	// frequently, it means that our cache size is too small).
+	evictCallback := func(key interface{}, value interface{}) {
+		cacheKey, ok := key.(CacheKey)
+		if !ok {
+			return
+		}
+		kp, err := cacheKey.toCacheKeyParts()
+		if err != nil {
+			return
+		}
+
+		identifierParts := strings.Split(kp.Identifier, "/")
+		if len(identifierParts) == 0 {
+			return
+		}
+
+		// Only increment the metric if we find an org _and_ repo.
+		// If the identifier begins with https://... then it's for Gerrit. In
+		// this case we need to have at least 4 parts (because we expect 2
+		// slashes for the protocol and at least 1 more slash to separate the
+		// "org" from the repo).
+		if identifierParts[0] == "http:" || identifierParts[0] == "https:" {
+			if len(identifierParts) < 4 {
+				return
+			}
+		} else {
+			// For GitHub, the format is just "org/repo" so there should be just
+			// 2 pieces.
+			if len(identifierParts) != 2 {
+				return
+			}
+		}
+
+		orgParts := identifierParts[:len(identifierParts)-1]
+		org := strings.Join(orgParts, "/")
+		repo := identifierParts[len(identifierParts)-1]
+		evictionsMetric.WithLabelValues(component, org, repo).Inc()
```

Comment:
This will make the repo the last part that doesn't contain a slash which is incorrect. Repos can contain slashes. 
Can we reuse/generalize some logic for getting the org and repo from the identifier rather than doing this by hand here?